neuralmonkey/tf_utils.py

```diff
@@ -5,34 +5,27 @@
 # pylint: enable=unused-import
 import tensorflow as tf
 
-from neuralmonkey.logging import debug
 
-
-# pylint: disable=invalid-name
-_initializers = {}  # type: Dict[str, Callable]
-_initialized_variables = set()  # type: Set[str]
-# pylint: enable=invalid-name
+def _get_current_experiment():
+    # This is needed to avoid circular imports.
+    from neuralmonkey.experiment import Experiment
+    return Experiment.get_current()
 
 
 def update_initializers(initializers: Iterable[Tuple[str, Callable]]) -> None:
-    _initializers.update(initializers)
+    _get_current_experiment().update_initializers(initializers)
 
 
 def get_initializer(var_name: str,
                     default: Callable = None) -> Optional[Callable]:
-    """Return the initializer associated with the given variable name."""
-    full_name = tf.get_variable_scope().name + "/" + var_name
-    initializer = _initializers.get(full_name, default)
-    if initializer is not default:
-        debug("Using {} for variable {}".format(initializer, full_name))
-    _initialized_variables.add(full_name)
-    return initializer
+    """Return the initializer associated with the given variable name.
 
+    The name of the current variable scope is prepended to the variable name.
 
-def get_unused_initializers() -> List[str]:
-    """Return the names of unused initializers."""
-    return [name for name in _initializers
-            if name not in _initialized_variables]
+    This should only be called during model building.
```

Comment:
Máme jak to tady zjistit přes ten current experiment, abychom se nespoléhali jen na to, že lidi čtou dokumentaci?

Reply:
Když to někdo zavolá mimo, tak to odchytí ten [`_DummyExperiment`](https://github.com/ufal/neuralmonkey/blob/234b634c3aeebcfbca30dfde3521bdb1f76394aa/neuralmonkey/experiment.py#L354), který na to upozorní.